Secret-agency-web/src/main/java/cz/fi/muni/pa165/secretagency/controllers/MissionsController.java

```diff
@@ -0,0 +1,181 @@
+package cz.fi.muni.pa165.secretagency.controllers;
+
+import cz.fi.muni.pa165.secretagency.ApiUris;
+import cz.fi.muni.pa165.secretagency.dto.MissionCreateDTO;
+import cz.fi.muni.pa165.secretagency.dto.MissionDTO;
+import cz.fi.muni.pa165.secretagency.enums.MissionTypeEnum;
+import cz.fi.muni.pa165.secretagency.exceptions.InvalidDeleteRequestException;
+import cz.fi.muni.pa165.secretagency.exceptions.ResourceNotFoundException;
+import cz.fi.muni.pa165.secretagency.facade.MissionFacade;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * REST controller for missions.
+ *
+ * @author Milos Silhar (433614)
+ */
+@RestController
+@RequestMapping(ApiUris.ROOT_URI_REST + ApiUris.ROOT_URI_MISSIONS)
+public class MissionsController {
+    final  static Logger logger = LoggerFactory.getLogger(MissionsController.class);
+
+    @Autowired
+    private MissionFacade missionFacade;
+
+    /**
+     * Get list of all missions
+     * curl --cookie {COOKIE_VALUE} -i -X GET http://localhost:8080/pa165/rest/missions
+     *
+     * @return List of all missions
+     */
+    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
Já vím, že jsi to viděl u dalších Controllerů, ale produces psát nemusíš. Když je třída označená jeko RestController, tak automaticky vrací JSON. Opravovat to nemusíš, píšu to jen pro info.